File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        String a[]=s.trim().split(" ");
      StringBuilder sb=new StringBuilder();
      for(int i=a.length-1;i>=0;i--){
        String st=a[i];
        if(st.trim().length()>0){
            sb.append(st.trim());
            sb.append(" ");
        }
      }
      return sb.toString().trim();
    }
}

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        t=s.strip().split(' ')
        ans=[]
        for i in range(len(t)-1,-1,-1):
            st=t[i]
            if st.strip():
                ans.append(st)
        return " ".join(ans)

        